packages/features/webhooks/lib/triggerDelegationCredentialErrorWebhook.ts

```diff
@@ -0,0 +1,89 @@
+import type { CalendarAppDelegationCredentialError } from "@calcom/lib/CalendarAppError";
+import logger from "@calcom/lib/logger";
+import { WebhookTriggerEvents } from "@calcom/prisma/enums";
+
+import { WebhookRepository } from "./repository/WebhookRepository";
+import sendPayload from "./sendPayload";
+
+const log = logger.getSubLogger({ prefix: ["triggerDelegationCredentialErrorWebhook"] });
+
+/**
+ * Triggers delegation credential error webhooks for organization webhooks subscribed to DELEGATION_CREDENTIAL_ERROR
+ */
+export async function triggerDelegationCredentialErrorWebhook(params: {
+  error: CalendarAppDelegationCredentialError;
+  credential: {
+    id: number;
+    type: string;
+    appId: string | null;
+  };
+  user: {
+    id: number;
+    email: string;
+  };
+  orgId?: number | null;
+}): Promise<void> {
+  try {
+    const { error, credential, user, orgId } = params;
+
+    if (!orgId) {
+      log.debug("Skipping webhook emission - no organization context");
+      return;
+    }
+
+    const webhookRepository = WebhookRepository.getInstance();
+    const webhooks = await webhookRepository.getSubscribers({
+      teamId: orgId,
+      triggerEvent: WebhookTriggerEvents.DELEGATION_CREDENTIAL_ERROR,
+    });
+
+    if (webhooks.length === 0) {
+      log.debug("No webhooks subscribed to DELEGATION_CREDENTIAL_ERROR for organization", { orgId });
+      return;
+    }
+
+    const payload = {
```

Comment:
We can strongly type it `const payload: DelegationCredentialErrorPayloadType = {`